lib/rules/utils/ast-utils.js

```diff
@@ -1006,6 +1006,42 @@ function isTopLevelExpressionStatement(node) {
 
 }
 
+/**
+ * Check whether the given node is a string literal not surrounded by parentheses.
+ * @param {ASTNode} node The node to check.
+ * @param {SourceCode} sourceCode The source code object to get tokens.
+ * @returns {boolean} `true` if the node is a string literal not surrounded by parentheses.
+ */
+function isUnparenthesizedStringLiteral(node, sourceCode) {
+    return node.type === "Literal" && typeof node.value === "string" && !isParenthesised(sourceCode, node);
+}
+
+/**
+ * Check whether the given node is a part of a directive prologue or not.
+ * @param {ASTNode} node The node to check.
+ * @param {SourceCode} sourceCode The source code object to get tokens.
+ * @returns {boolean} `true` if the node is a part of directive prologue.
+ */
+function isDirective(node, sourceCode) {
+    if (node.type === "ExpressionStatement" && "directive" in node) {
```

Comment:
Good catch! I think it's safe to assume that if `directive` is `null` then the node is not a directive, even if we decide to keep the fallback logic. I've pushed an update in https://github.com/eslint/eslint/commit/84e71335a56914e05c18a30ade8cc88c14b55af9, but somehow the commit is not showing up in the PR...

Reply:
> I've pushed an update in [84e7133](https://github.com/eslint/eslint/commit/84e71335a56914e05c18a30ade8cc88c14b55af9), but somehow the commit is not showing up in the PR...

Weird, this PR's branch indeed points to https://github.com/eslint/eslint/commit/84e71335a56914e05c18a30ade8cc88c14b55af9: https://github.com/eslint/eslint/tree/issue16988-no-useless-escape

It could be some temporary GitHub UI issue.

Reply:
> It could be some temporary GitHub UI issue.

Yeah, after an empty commit the PR is now up to date.